### PR TITLE
rewrite README.md

### PR DESCRIPTION
**Pull request by chdoc**
This is an attempt at rewriting the `README.md` file to take into account some of the discussion of the [CUDW Working Group](https://coq.zulipchat.com/#narrow/stream/255971-CUDW-2020/topic/WG.3A.20coq-community.20template.20documentation) 

closes: #78

### CODE REVIEW

**Comment by chdoc** [comment]
Note, the rendering of the `README.md` is available here: https://github.com/coq-community/templates/tree/README

File: README.md [hunk 1]
```diff
@@ -30,26 +30,26 @@ YAML-compatible mustache implementation such as `ruby-mustache`
...
+one of the following projects may serve as a starting point:
+
+- [Huffman](https://github.com/coq-community/huffman) (pure library, no dependencies, coq-community) :
+  [meta.yml](https://github.com/coq-community/huffman/blob/master/meta.yml)
+- [AAC_tactics](https://github.com/coq-community/aac-tactics) (plugin, one-branch per Coq version):
+  [meta.yml](https://github.com/coq-community/aac-tactics/blob/master/meta.yml).
+- [reglang](https://github.com/coq-community/reglang) (pure library, mathcomp, custom docker images) :
+  [meta.yml](https://github.com/coq-community/reglang/blob/master/meta.yml).
+- [StructTact](https://github.com/uwplse/StructTact) (pure library, no dependencies, no coq-community):
+  [meta.yml](https://github.com/uwplse/StructTact/blob/master/meta.yml)
```
Comment (palmskog) [inline README.md:45]:
```suggestion
- [Huffman](https://github.com/coq-community/huffman) (pure library, no dependencies, coq-community):
  [meta.yml](https://github.com/coq-community/huffman/blob/master/meta.yml)
- [AAC Tactics](https://github.com/coq-community/aac-tactics) (plugin, one-branch per Coq version):
  [meta.yml](https://github.com/coq-community/aac-tactics/blob/master/meta.yml)
- [reglang](https://github.com/coq-community/reglang) (pure library, mathcomp, custom docker images):
  [meta.yml](https://github.com/coq-community/reglang/blob/master/meta.yml)
- [StructTact](https://github.com/uwplse/StructTact) (pure library, no dependencies, no coq-community):
  [meta.yml](https://github.com/uwplse/StructTact/blob/master/meta.yml)
```
Nitpicks.

Comment (chdoc) [reply README.md:45]: Nitpicking is accepted, even welcome, in this context.

File: README.md [hunk 2]
```diff
@@ -67,14 +67,32 @@ Regarding continuous integration, the `generate.sh` script will create:
 depending on whether `meta.yml` contains `travis: true` or `action: true`
 or `circleci: true`.
 
-If you only want to generate certain files, please specify them as shell script arguments:
+For `coq-community` projects, using `travis: true` is currently
+disabled due to a very limed build-time allowance from Travis. Using
```
Comment (Zimmi48) [inline README.md:71]:
```suggestion
disabled due to a very limited build-time allowance from Travis. Using
```

File: README.md [hunk 3]
```diff
@@ -67,14 +67,32 @@ Regarding continuous integration, the `generate.sh` script will create:
...
+also be generated by calling `mustache` directly.  Asuming the
+templates are available in a sibling directory, the configuration and
+boilerplate files for [AAC_tactics](https://github.com/coq-community/aac-tactics) are
+generated as follows:
+
+```shell
+mustache meta.yml ../templates/README.md.mustache > README.md
+mustache meta.yml ../templates/coq.opam.mustache > coq-aac-tactics.opam
+mustache meta.yml ../templates/default.nix.mustache > default.nix
+mustache meta.yml ../templates/.travis.yml.mustache > .travis.yml
```
Comment (Zimmi48) [inline README.md:89]: Maybe reduce this example to just the `README`? This is not a very frequent use case anyway, and the list of file is already out-of-date (Travis).

File: README.md [hunk 4]
```diff
@@ -67,14 +67,32 @@ Regarding continuous integration, the `generate.sh` script will create:
...
 ```
 
-Keeping generated files under version control is not ideal, but `README.md` has to exist,
-and generally this is a common practice when using build systems such as Autotools.
-To get a `mustache` tool in, e.g., NixOS, you can run `nix-env -i mustache-go`.
+The generated files should be kept under version control, as
+continuous integration needs its config file as well as the `.opam`
+file to be present in the repository and and GitHub needs the
```
Comment (Zimmi48) [inline README.md:94]:
```suggestion
file to be present in the repository and GitHub needs the
```

**Review by Zimmi48 — APPROVED** [review]
Looks great!

File: README.md (hunk 3 above)
Comment (chdoc) [reply README.md]: @Zimmi48 done! I also added some headlines to structure the text and move this alternative out of the main flow.

**Review by palmskog — APPROVED** [review]
I think this is ready to merge now, if @chdoc agrees.

**Comment by chdoc** [comment]
@palmskog Sure, go ahead and merge. I was hoping for some feedback from @amahboubi , whose difficulties in using the templates prompted the rewrite. But since we all seem to be in agreement that this is an improvement over the current state, any further improvements can be done at a later point :grin:

**Pull request merged**